reducers/chatsReducer.js

```diff
@@ -14,38 +14,56 @@ function chatsReducer(state = chatsInitialState, action) {
                 ...state,
                 chats: payload
             };
+
         case types.SET_CURRENT_CHAT:
             return {
                 ...state,
                 currentChat: payload
             };
+
         case types.CREATE_CHAT:
             return {
                 ...state,
                 chats: state.chats.concat(payload)
             };
+
         case types.CREATE_USER_IN_CHAT:
             const { id, display_name: displayName, cid } = payload;
-            const { chats } = state;
+            
             const user = {
                 id,
                 displayName
             };
-            // Concat user object to the users array of assoc. chat
-            const cidIndex = chats.findIndex(
-                (item) => item.id === cid
+
+            const cidIndex = state.chats.findIndex(
+                (chat) => chat.id === cid
             );
+
+            // make copy of chat to update
+            const chatToUpdate = Object.assign({}, state.chats[cidIndex]);
+            // concat added user and return new array of users
+            const updatedChatUsers = { users: chatToUpdate.users.concat(user) };
+            // make copy of updated chat to replace given chat in the state
+            const updatedChat = { ...chatToUpdate, ...updatedChatUsers };
+
             return {
                 ...state,
                 addedUser: user,
-                chats: {
-                    ...state.chats,
-                    [chats[cidIndex].users]: [
-                        ...state.chats[cidIndex].users,
-                        user
-                    ]
-                }
+                chats: [
+                    ...state.chats.slice(0, cidIndex),
+                    updatedChat,
+                    ...state.chats.slice(cidIndex + 1)
+                ]
```

Comment:
yeah, so another thing i noticed is chats are not the right datatype :/ and that originated in [his previous pr](https://github.com/emote-chat/client/pull/10/files). updated so it's an array and i followed the [documentation](https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns) on how to update the chat at the given index. i wanted to use map here with a ternary operator, but i wasn't sure what was general/best practice here from a react standpoint. lmk if you have any particular opinion either way.